source/locationHelper.py

```diff
@@ -0,0 +1,402 @@
+			except ValueError:
+				return False
+		return (self.y, self.x) >= (other.y, other.x)
+
+	def __eq__(self, other):
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return self.x==other.x and self.y==other.y
+
+	def __neq__(self, other):
+		if not isinstance(other,_POINT_CLASSES):
+			try:
+				other=toPoint(other)
+			except ValueError:
+				return False
+		return self.x!=self.x or self.y!=other.y
+
+	def __sub__(self,other):
+		if not isinstance(other,_POINT_CLASSES):
+			return NotImplemented
+		return Point((self.x-other.x),(self.y-other.y))
+
+	def __rsub__(self,other):
+		return self.__sub__(other)
+
+	def toPOINT(self):
+		"""Converts self to a L{ctypes.wintypes.POINT}"""
+		return POINT(self.x,self.y)
+
+	def toLogical(self, hwnd):
+		"""Converts self from physical to logical coordinates and returns a new L{Point}."""
+		return Point(*windowUtils.physicalToLogicalPoint(hwnd, *self))
+
+	def toPhysical(self, hwnd):
+		"""Converts self from logical to physical coordinates and returns a new L{Point}"""
+		return Point(*windowUtils.logicalToPhysicalPoint(hwnd, *self))
+
+	def toClient(self, hwnd):
+		"""Converts self from screen to client coordinates and returns a new L{Point}"""
+		return Point(*winUser.ScreenToClient(hwnd, *self))
+
+	def toScreen(self, hwnd):
+		"""Converts self from client to screen coordinates and returns a new L{Point}"""
+		return Point(*winUser.ClientToScreen(hwnd, *self))
+
+class _RectMixin:
+	"""Mix-in class for properties shared between Location and Rect classes"""
+
+	def toRECT(self):
+		"""Converts self to a L{ctypes.wintypes.RECT}"""
+		return RECT(self.left,self.top,self.right,self.bottom)
+
+	def toLogical(self, hwnd):
+		left,top=self.topLeft.toLogical(hwnd)
+		right,bottom=self.bottomRight.toLogical(hwnd)
+		if type(self) is Location:
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def toPhysical(self, hwnd):
+		left,top=self.topLeft.toPhysical(hwnd)
+		right,bottom=self.bottomRight.toPhysical(hwnd)
+		if type(self) is Location:
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def toClient(self, hwnd):
+		left, top =self.topLeft.toClient(hwnd)
+		if type(self) is Location:
+			return Location(left, top, self.width, self.height)
+		return Rect(left, top, left+self.width, top+self.height)
+
+	def toScreen(self, hwnd):
+		left,top=self.topLeft.toScreen(hwnd)
+		if type(self) is Location:
+			return Location(left, top, self.width, self.height)
+		return Rect(left, top, left+self.width, top+self.height)
+
+	@property
+	def topLeft(self):
+		return Point(self.left,self.top)
+
+	@property
+	def bottomRight(self):
+		return Point(self.right,self.bottom)
+
+	@property
+	def center(self):
+		return Point((self.left+self.right)/2, (self.top+self.bottom)/2)
+
+	def __contains__(self,other):
+		"""Returns whether other is a part of self."""

```

Comment:
While this literally says what the function's code does, I would change "part of self" to something like "part of this rectangle"